### PR TITLE
Sendemail

### PR DESCRIPTION
**Pull request by franzmoro**
### Tasks
- [x] setup sendemail module
- [x] send email upon lecturer verification (incl. adjusting db, and visualizing messages)
- [x] update db fields upon lecturer visiting verification url
- [x] prevent lecturer access unless email is verified

related #251, #252

### CODE REVIEW

File: server/handlers/save-user.js [hunk 1]
```diff
@@ -12,26 +14,43 @@ module.exports = {
         var password = request.payload.password;
         var is_lecturer = request.payload.is_lecturer;
         var username = request.payload.username || '';
+        var verification_code = is_lecturer ? uuid() : null;
 
         getUserByEmail(client, email, (error, userExists) => {
```
Comment (Danwhy) [inline server/handlers/save-user.js]: We've missed handling the error here

File: server/handlers/save-user.js [hunk 2]
```diff
@@ -12,26 +14,43 @@ module.exports = {
...
                         }
-                        getUserByEmail(client, email, (error, userDetails) => {
-                            delete userDetails[0].password;
-                            return reply(userDetails[0])
-                            .state('cul_id', userDetails[0].user_id.toString(), { path: "/" })
-                            .state('cul_is_lecturer', userDetails[0].is_lecturer.toString(), { path: "/" })
-                            .state('cul_is_cookie_accepted', 'true', { path: "/" });
-                        });
+                        else if (!is_lecturer) {
+                            getUserByEmail(client, email, (error, userDetails) => {
```
Comment (Danwhy) [inline server/handlers/save-user.js]: same here

**Comment by iteles** [comment]
<3 comments on code coverage, thanks for the review @Danwhy !

File: server/handlers/save-user.js (hunk 2 above)
Comment (franzmoro) [inline server/handlers/save-user.js]: addressed

File: server/handlers/save-user.js (hunk 1 above)
Comment (franzmoro) [inline server/handlers/save-user.js]: addressed

**Comment by franzmoro** [comment]
@Danwhy do you have a good example of Sinon usage (e.g. a project that uses it)?

**Comment by Danwhy** [comment]
I used it [another project here](https://github.com/TheScienceMuseum/collectionsonline/blob/831abcbd992729eef902748248c3d42b5215c6ea/test/error.test.js#L147). You call Sinon's stub method, passing it the module, the name of the module's method you want to mock, and the function you want to replace it with. 

In that example I'm mocking the cache's 'get' method so it always returns an error in a callback, which lets me test how the error is being handled.

**Comment by franzmoro** [comment]
hey @Danwhy coverage on the diff is now up to 100%. re-assigning the PR to you. 😉

**Comment by franzmoro** [comment]
thanks, @Danwhy & obviously sharing the 🥇 with @sohilpandya 😄

**Pull request merged**